### PR TITLE
Functional login UI with transcript output

### PR DESCRIPTION
**Pull request by msoechting**

### CODE REVIEW

**Comment by querenker** [comment]
At a first glance this code looks very good, but I have some questions resp. annotations:
- The login method is very long. Maybe it is a good idea to split it? For example the connect stuff and the check for an successful connect could in each case extracted to a new method?
- Are the default values for the connection on UI level necessary? Maybe they are better placed at endpoint level?
- the methods open, buildWith and login are as yet unclassified

**Comment by msoechting** [comment]
Thanks for the feedback!
- Will change that.
- I think it's a good idea to have the default values on the UI level as their only purpose is to tell the user which field contains what information. (The values have no internal meaning and are (in my opinion) equivalent to a label next to the input field which you would hard-code on UI level.)
- Will fix.

**Comment by querenker** [comment]
If the default values only act as label then it is all right 👍

File: packages/IMAPClient-UI.package/ICLoginDialog.class/instance/buildWith..st [hunk 1]
```diff
@@ -47,12 +50,14 @@ buildWith: builder
...
-		frame: (self guiElementXOffset @ (self guiElementYOffset + 0.46) extent: self guiElementWidth @ self guiElementHeight).
+		help: #passwordHelpText;
+		font: (StrikeFont passwordFontSize: 12);
+		frame: (self guiElementXOffset @ (self guiElementYOffsetOf: 3) extent: self guiElementWidth @ self guiElementHeight).
 	
 	loginButtonSpec := builder pluggableButtonSpec new.
 	loginButtonSpec
 		model: self;
-		frame: (self guiElementXOffset @ (self guiElementYOffset + 0.7) extent: self guiElementWidth @ (self guiElementHeight * 1.3));
+		frame: (self guiElementXOffset @ (self guiElementYOffsetOf: 4.2) extent: self guiElementWidth @ (self guiElementHeight * 1.3));
```
Comment (julianweise) [inline packages/IMAPClient-UI.package/ICLoginDialog.class/instance/buildWith..st:60]: 1.3 magic number?
maybe outsource Font + Fontsize to class site

**Pull request merged**